### DISCUSSION TITLE
How to parse (serialize) an image added in a post

### DISCUSSION THREAD

**Discussion by mikevarela** [payloadcms/payload#1247] [Feature Requests & Ideas]
I've built a posts collection. Have added upload to the collection to 'Media' and now have a main image used for the summary card and the main post page image (featured). However, when writing a post I'd like to add images throughout. I know you have add and upload planned for a future release, that's great! However, I'm trying to figure out how to display this image in my page. I've checked Slates docs and your own and looked at the serialize ts file and dont' see where to parse and display using the RichText component. Can you help?

**Comment by unonweb** [comment]
Hi, I'm using a modified version of the function mentioned [here](https://payloadcms.com/docs/fields/rich-text#generating-html) which is called by the afterChange hook. The function is missing a case for `node.type === 'upload'`. So I've added it:

```js
if (node) {
  let nodeHTML
  switch (node.type)
    case 'upload':
      nodeHTML = _createImgSrcSetEl(node.value, webImgDir)
      //`<strong>${node.relationTo} Upload: ${node.value}</strong><br/>`;
      break;
  //...
}
```
By `_createImgSrcSetEl()` an `<img>` s returned and added to the whole html string.
So in the end I save a server side created html string into the json for later use. If you any more questions or need the whole code involved, feel free to ask.
